eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/PointerStatements2.scala

```diff
@@ -0,0 +1,75 @@
+package com.evolutiongaming.kafka.journal.eventual.cassandra
+
+import cats.data.{NonEmptyList => Nel}
+import cats.syntax.all._
+import cats.{Monad, Parallel}
+import com.evolutiongaming.kafka.journal.eventual.cassandra.CassandraHelper._
+import com.evolutiongaming.kafka.journal.eventual.cassandra.EventualCassandraConfig.ConsistencyConfig
+import com.evolutiongaming.scassandra.TableName
+import com.evolutiongaming.scassandra.syntax._
+import com.evolutiongaming.skafka.{Partition, Topic}
+
+object PointerStatements2 {
+
+  def createTable(name: TableName): String = {
+    s"""
+       |CREATE TABLE IF NOT EXISTS ${ name.toCql } (
+       |topic text,
+       |partition int,
+       |offset bigint,
+       |created timestamp,
+       |updated timestamp,
+       |PRIMARY KEY ((topic, partition)))
+       |""".stripMargin
+  }
+
+  object SelectIn {
+
+    def of[F[_]: Monad: Parallel: CassandraSession](
+      name: TableName,
+      consistencyConfig: ConsistencyConfig.Read
+    ): F[PointerStatements.SelectIn[F]] = {
+
+      PointerStatements.Select
+        .of[F](name, consistencyConfig)
+        .map { select =>
+          (topic: Topic, partitions: Nel[Partition]) => {
+            partitions
+              .toList
+              .parFlatTraverse { partition =>
+                select(topic, partition).map { offset =>
+                  offset.toList.map { offset => partition -> offset }
+                }
+              }
+              .map(_.toMap)
+          }
+        }
+
+    }
+  }
+
+  object SelectTopics {
+
+    def of[F[_]: Monad: CassandraSession](
+      name: TableName,
+      consistencyConfig: ConsistencyConfig.Read
+    ): F[PointerStatements.SelectTopics[F]] = {
+
+      val query = s"""SELECT DISTINCT topic, partition FROM ${ name.toCql }""".stripMargin
```

Comment:
do you need `partition` here?

Reply:
Cassandra does not allow distinct queries of non full primary key